specutils/spectra/xrayspectrum1d.py

```diff
@@ -0,0 +1,542 @@
+            ARF), this keyword provides the functionality to override the
+            default behaviour and manually set the exposure time to use.
+
+        store_model_counts : bool
+            If True, the output will also be stored in self.model_counts
+
+        Returns
+        -------
+        model_counts : numpy.ndarray
+            The model spectrum in units of counts/bin
+
+        If no ARF file exists, it will return the model flux after applying the RMF
+        If no RMF file exists, it will return the model flux after applying the ARF (with a warning)
+        If no ARF and no RMF, it will return the model flux spectrum (with a warning)
+        """
+
+        if self.arf is not None:
+            mrate  = self.arf.apply_arf(mflux, exposure=exposure)
+        else:
+            #print("Caution: no ARF file specified")
+            mrate = mflux
+
+        if self.rmf is not None:
+            result = self.rmf.apply_rmf(mrate)
+        else:
+            #print("Caution: no RMF file specified")
+            result = mrate
+
+        return result
+
+## ----  Supporting response file objects
+
+class RMF(object):
+    def __init__(self, filename):
+        self._load_rmf(filename)
+        pass
+
+    def _load_rmf(self, filename):
+        """
+        Load an RMF from a FITS file.
+
+        Parameters
+        ----------
+        filename : str
+            The file name with the RMF file
+
+        Attributes
+        ----------
+        filename : str
+            The file name that the RMF was drawn from
+
+        n_grp : numpy.ndarray
+            the Array with the number of channels in each
+            channel set
+
+        f_chan : numpy.ndarray
+            The starting channel for each channel group;
+            If an element i in n_grp > 1, then the resulting
+            row entry in f_chan will be a list of length n_grp[i];
+            otherwise it will be a single number
+
+        n_chan : numpy.ndarray
+            The number of channels in each channel group. The same
+            logic as for f_chan applies
+
+        matrix : numpy.ndarray
+            The redistribution matrix as a flattened 1D vector
+
+        energ_lo : numpy.ndarray
+            The lower edges of the energy bins
+
+        energ_hi : numpy.ndarray
+            The upper edges of the energy bins
+
+        energ_unit : astropy.units.Unit
+            Description of the energy units used
+
+        detchans : int
+            The number of channels in the detector
+
+        """
+        # open the FITS file and extract the MATRIX extension
+        # which contains the redistribution matrix and
+        # anxillary information
+        hdulist = fits.open(filename)
+        self.filename = filename
+
+        # get all the extension names
+        extnames = np.array([h.name for h in hdulist])
+
+        # figure out the right extension to use
+        if "MATRIX" in extnames:
+            h = hdulist["MATRIX"]
```

Comment:
As suggestion, make "MATRIX" a variable here so it isn't hard coded then the user can specify what the keyword for this is.  It could default to MATRIX

Reply:
The user should not ever have to think about this, and the extension depends on the instrument or data reduction tool used to produce the RMF.

For now, I am adding an "extension" keyword that defaults to None. The `_load_rmf` will check if the user has provided a value, and use it. Otherwise, it searches for "MATRIX" or "SPECRESP MATRIX". If extension=None, and the two common extensions are not found, a message is printed and the function quits.